sdk/src/main/java/com/robotemi/sdk/Robot.kt

```diff
@@ -3267,6 +3333,109 @@ class Robot private constructor(private val context: Context) {
         return mapDataModel
     }
 
+    /**
+     * Get map elements, such as locations, virtual walls, green paths, etc.
+     *
+     * @return Map elements.
+     */
+    fun getMapElements(): List<Layer>? {
+        if (checkSelfPermission(Permission.MAP) == Permission.DENIED) {
+            Log.e(TAG, "getMapElements() - Permission denied")
+            return null
+        }
+//        if (isMapLocked() == true) return sdkService?.getMapElements(applicationInfo.packageName)
+        var cursor: Cursor? = null
+        val uriStr = StringBuffer("content://")
+            .append(SdkConstants.PROVIDER_AUTHORITY)
+            .append("/").append(SdkConstants.PROVIDER_PARAMETER_MAP_DATA)
+            .toString()
+        cursor = context.contentResolver.query(
+            Uri.parse(uriStr),
+            null,
+            null,
+            null,
+            null
+        )
+        if (cursor == null || !cursor.moveToFirst()) {
+            return null
+        }
+        val mapElementsJson = cursor.getString(cursor.getColumnIndexOrThrow(MAP_ELEMENTS))
+        cursor.close()
+        return gson.fromJson<List<Layer>>(
+            mapElementsJson,
+            object : TypeToken<List<Layer>>() {}.type
+        )
+    }
+
+    /**
+     * Get map image
+     *
+     * @return Map image
+     */
+    fun getMapImage(): MapImage? {
```

Comment:
Let's use MapDataModel as response and declare it only contains MapImage and MapInfo